### ISSUE TITLE
ArgoCD azureAD authentication failed

### ISSUE THREAD

**Issue by dbacademy** [argoproj/argo-cd#13606]
Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

I have installed argocd:v2.7.2 into my Openshift cluster; but authentication is failing, unable to get the token as per the logs 

this is my configuration of arcdcd-cm ConfigMap

data:
  oidc.config: >
    name: AzureAD   

    issuer:
    https://login.microsoftonline.com/81fa766e-a349-4867-8bf4-ab35e250a08f/v2.0

    clientID: jhjfhjhjdd-kjfkjkjkf-uiyiyir-ijiohjihjh

    clientSecret: xxxx~xxxxxxxxxxxxxx~_xxxxxxxxxxxxxx~wssxxxxxxxxxxx

    requestedIDTokenClaims:
        groups:
            essential: true
    requestedScopes:
            - openid
            - profile
            - email
  url: 'https://argocd-server-argocd.apps.mq-azureocp-nonprod.volvocars.net/'

**To Reproduce**

**Expected behavior**

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

023-05-16T10:50:25Z" grpc.time_ms=27.273 span.kind=server system=grpc
time="2023-05-16T10:50:25Z" level=info msg="Initializing OIDC provider (issuer: https://login.microsoftonline.com/81fa766e-a349-4867-8bf4-ab35e250a08f/v2.0)"
time="2023-05-16T10:50:25Z" level=warning msg="Failed to verify token: failed to verify token: Failed to query provider \"https://login.microsoftonline.com/81fa766e-a349-4867-8bf4-ab35e250a08f/v2.0\": Get \"https://login.microsoftonline.com/81fa766e-a349-4867-8bf4-ab35e250a08f/v2.0/.well-known/openid-configuration\": x509: certificate signed by unknown authority"
time="2023-05-16T10:50:25Z" level=info msg="received unary call /cluster.SettingsService/Get" grpc.method=Get grpc.request.content= grpc.service=cluster.SettingsService grpc.start_time="2023-05-16T10:50:25Z" span.kind=server system=grpc

**Comment by woniuzhang** [comment]
I had the same issue, did you solve it?

**Comment by rickardk-se** [comment]
Same error here, any ideas?

**Comment by Vinaum8** [comment]
this link maybe can help: https://argo-cd.readthedocs.io/en/stable/faq/#why-am-i-getting-x509-certificate-signed-by-unknown-authority-when-using-the-cli

**Comment by dbacademy** [comment]
> Same error here, any ideas?

yes .. it was resolved .. following three you need to grant for application permission instead of delegated 

Group.Read.All
GroupMember.Read.All
User.Read.All

**Comment by dbacademy** [comment]
> I had the same issue, did you solve it?

yes .. it was resolved .. following three you need to grant for application permission instead of delegated 

Group.Read.All
GroupMember.Read.All
User.Read.All
